tests/client/bulk.rs

```diff
@@ -0,0 +1,73 @@
+use bson::{Bson, Document};
+use mongodb::client::coll::options::WriteModel;
+use mongodb::client::MongoClient;
+
+#[test]
+fn bulk_ordered_insert_only() {
+    let client = MongoClient::new("localhost", 27017).unwrap();
+    let db = client.db("test");
+    let coll = db.collection("bulk_ordered_insert_only");
+
+    coll.drop().unwrap();
+
+    let models = (1..5).map(|i| WriteModel::InsertOne { document: doc! {
+        "_id" => (i),
+        "x" => (i * 11)
+    }}).collect();
+
+    coll.bulk_write(models, true);
+
+    let cursor : Vec<_> = coll.find(None, None).unwrap().collect();
+
+    assert_eq!(cursor.len(), 4);
+
+    for (i, result) in cursor.into_iter().enumerate() {
+        let doc = result.unwrap();
+        let i = i + 1;
```

Comment:
Since `i` is the index, it starts at 0, but the `_id` fields start at one. I just figured the easiest way to deal with this was to add one to it beforehand rather than using `i + 1` twice, since the need to case in both cases and the need to multiply in the second would make the additional operation a bit messy.